core/indexer/interface.go

```diff
@@ -2,44 +2,66 @@ package indexer
 
 import (
 	"bytes"
-	"io"
 
+	"github.com/ElrondNetwork/elrond-go/core/indexer/workItems"
 	"github.com/ElrondNetwork/elrond-go/core/statistics"
 	"github.com/ElrondNetwork/elrond-go/data"
 	"github.com/ElrondNetwork/elrond-go/data/block"
 	"github.com/ElrondNetwork/elrond-go/process"
 	"github.com/elastic/go-elasticsearch/v7/esapi"
 )
 
+// DataIndexerFactory can create new instances of Indexer
+type DataIndexerFactory interface {
+	Create() (Indexer, error)
+	IsInterfaceNil() bool
+}
+
 // Indexer is an interface for saving node specific data to other storage.
 // This could be an elastic search index, a MySql database or any other external services.
 type Indexer interface {
 	SetTxLogsProcessor(txLogsProc process.TransactionLogProcessorDatabase)
 	SaveBlock(body data.BodyHandler, header data.HeaderHandler, txPool map[string]data.TransactionHandler, signersIndexes []uint64, notarizedHeadersHashes []string)
-	SaveRoundsInfos(roundsInfos []RoundInfo)
+	RevertIndexedBlock(header data.HeaderHandler, body data.BodyHandler)
+	SaveRoundsInfo(roundsInfos []workItems.RoundInfo)
 	UpdateTPS(tpsBenchmark statistics.TPSBenchmark)
 	SaveValidatorsPubKeys(validatorsPubKeys map[uint32][][]byte, epoch uint32)
-	SaveValidatorsRating(indexID string, infoRating []ValidatorRatingInfo)
+	SaveValidatorsRating(indexID string, infoRating []workItems.ValidatorRatingInfo)
+	StopIndexing() error
 	IsInterfaceNil() bool
 	IsNilIndexer() bool
 }
 
-// DatabaseHandler is an interface used by elasticsearch component to prepare data to be saved on elasticsearch server
-type DatabaseHandler interface {
+// DispatcherHandler is an interface for the data dispatcher data is used to index data in database
+type DispatcherHandler interface {
+	StartIndexData()
+	Close() error
+	Add(item workItems.WorkItemHandler)
+}
+
+// ElasticProcessor defines the interface for the elastic search indexer
+type ElasticProcessor interface {
```

Comment:
IsInterfaceNil() method is not added here

Reply:
done.